src/main/java/org/reaktivity/nukleus/kafka/internal/cache/CompactedTopicCache.java

```diff
@@ -114,20 +123,41 @@ public MessageRef getMessage(
     }
 
     @Override
-    public void extendNextOffset(
+    public long getOffset(
         int partition,
-        long requestOffset,
-        long lastOffset)
+        OctetsFW key)
     {
-        indexes[partition].extendNextOffset(requestOffset, lastOffset);
+        return indexes[partition].getOffset(key);
     }
 
     @Override
-    public long getOffset(
-        int partition,
-        OctetsFW key)
+    public boolean hasMessages(
```

Comment:
This method is new, and that's the only significant change (I just moved another method in the file)